go/vt/vtctl/workflow/server.go

```diff
@@ -1292,6 +1300,218 @@ func (s *Server) ReshardCreate(ctx context.Context, req *vtctldatapb.ReshardCrea
 	return nil, nil
 }
 
+// VDiffCreate is part of the vtctlservicepb.VtctldServer interface.
+// It passes on the request to the target primary tablets that are
+// participating in the given workflow and VDiff.
+func (s *Server) VDiffCreate(ctx context.Context, req *vtctldatapb.VDiffCreateRequest) (*vtctldatapb.VDiffCreateResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "workflow.Server.VDiffCreate")
+	defer span.Finish()
+
+	span.Annotate("keyspace", req.TargetKeyspace)
+	span.Annotate("workflow", req.Workflow)
+	span.Annotate("source_cells", req.SourceCells)
+	span.Annotate("target_cells", req.TargetCells)
+	span.Annotate("tablet_types", req.TabletTypes)
+	span.Annotate("tables", req.Tables)
+	span.Annotate("auto_retry", req.AutoRetry)
+
+	tabletTypesStr := topoproto.MakeStringTypeCSV(req.TabletTypes)
+	if req.TabletSelectionPreference == tabletmanagerdatapb.TabletSelectionPreference_INORDER {
+		tabletTypesStr = discovery.InOrderHint + tabletTypesStr
+	}
+
+	options := &tabletmanagerdatapb.VDiffOptions{
+		PickerOptions: &tabletmanagerdatapb.VDiffPickerOptions{
+			TabletTypes: tabletTypesStr,
+			SourceCell:  strings.Join(req.SourceCells, ","),
+			TargetCell:  strings.Join(req.TargetCells, ","),
+		},
+		CoreOptions: &tabletmanagerdatapb.VDiffCoreOptions{
+			Tables:                strings.Join(req.Tables, ","),
+			AutoRetry:             req.AutoRetry,
+			MaxRows:               req.MaxExtraRowsToCompare,
+			TimeoutSeconds:        req.FilteredReplicationWaitTime.Seconds,
+			MaxExtraRowsToCompare: req.MaxExtraRowsToCompare,
+			UpdateTableStats:      req.UpdateTableStats,
+		},
+		ReportOptions: &tabletmanagerdatapb.VDiffReportOptions{
+			OnlyPks:    req.OnlyPKs,
+			DebugQuery: req.DebugQuery,
+		},
+	}
+
+	tabletreq := &tabletmanagerdatapb.VDiffRequest{
+		Keyspace:  req.TargetKeyspace,
+		Workflow:  req.Workflow,
+		Action:    string(vdiff.CreateAction),
+		Options:   options,
+		VdiffUuid: req.Uuid,
+	}
+
+	ts, err := s.buildTrafficSwitcher(ctx, req.TargetKeyspace, req.Workflow)
+	if err != nil {
+		return nil, err
+	}
+	if ts.frozen {
+		return nil, fmt.Errorf("invalid VDiff run: writes have been already been switched for workflow %s.%s",
```

Comment:
question for the whole file: should we be using vterrors here? (e.g. `FAILED_PRECONDITION` for this particular case)